oauth-server.php

```diff
@@ -40,6 +40,16 @@ function rest_oauth1_init() {
 
 	global $wp;
 	$wp->add_query_var('rest_oauth1');
+
+	// Detect HTTPS
+	if ( $_SERVER['HTTP_X_FORWARDED_PROTO'] == 'https' ) {
```

Comment:
I think this should really be in whatever project you are using this plugin in. `HTTP_X_FORWARDED_PROTO` is an environment level thing, where the parent project should have the responsibility of turning on 'https' is the traffic was forwarded from HTTPS. You'd need to do this elsewhere than this plugin, as WordPress it's self wouldn't realise you were on HTTPS for example.